app/src/main/java/com/artemzin/qualitymatters/network/HostSelectionInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package com.artemzin.qualitymatters.network;
+
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Such implementation allows us easily change base url in the integration and functional tests!
+ */
+@Singleton
```

Comment:
nooooooooooooooo please

Please instantiate object in the module explicitly, this makes it:

1. Overridable for tests/etc
2. Scope is explicitly visible because you have compontent + module relation